ixwebsocket/IXSocketOpenSSL.cpp

```diff
@@ -265,12 +268,53 @@ namespace ix
             }
 
             ERR_clear_error();
-            int cert_load_result = SSL_CTX_set_default_verify_paths(_ssl_context);
-            if (cert_load_result == 0)
-            {
-                unsigned long ssl_err = ERR_get_error();
-                errMsg = "OpenSSL failed - SSL_CTX_default_verify_paths loading failed: ";
-                errMsg += ERR_error_string(ssl_err, nullptr);
+            if (_tlsOptions.isUsingClientCert()) {
+                if (SSL_CTX_use_certificate_chain_file(_ssl_context, 
+                        _tlsOptions.cert_file.c_str()) != 1) {
+                    unsigned long ssl_err = ERR_get_error();
+                    errMsg = "OpenSSL failed - SSL_CTX_use_certificate_chain_file(\""
+                            + _tlsOptions.cert_file + "\") failed: ";
+                    errMsg += ERR_error_string(ssl_err, nullptr);
+                } else if (SSL_CTX_use_PrivateKey_file(_ssl_context, 
+                        _tlsOptions.key_file.c_str(), SSL_FILETYPE_PEM) != 1) {
+                    unsigned long ssl_err = ERR_get_error();
+                    errMsg = "OpenSSL failed - SSL_CTX_use_PrivateKey_file(\""
+                            + _tlsOptions.key_file + "\") failed: ";
+                    errMsg += ERR_error_string(ssl_err, nullptr);
+                }
+            }
+
+
+            ERR_clear_error();
+            if (!_tlsOptions.isPeerVerifyDisabled()) {
+            
+                if (_tlsOptions.isUsingSystemDefaults()) {
+                    int cert_load_result = SSL_CTX_set_default_verify_paths(_ssl_context);
+                    if (cert_load_result == 0)
+                    {
+                        unsigned long ssl_err = ERR_get_error();
+                        errMsg = "OpenSSL failed - SSL_CTX_default_verify_paths loading failed: ";
+                        errMsg += ERR_error_string(ssl_err, nullptr);
+                    }
+                } else {
+                    const char * root_ca_file = _tlsOptions.ca_file.c_str();
+                    STACK_OF(X509_NAME) *root_cas;
+                    root_cas = SSL_load_client_CA_file(root_ca_file);
+                    if (root_cas == NULL) {
```

Comment:
you could try: auto sslErr = ERR_get_error();

Reply:
ok